### PR TITLE
Improve checking the syntax for evaluator scripts

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt [hunk 1]
```diff
@@ -254,6 +239,19 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate rules
...
+            val outputFiles = mutableSetOf<File>()
+
+            outputFormats.mapTo(outputFiles) { format ->
+                absoluteOutputDir.resolve("evaluation-result.${format.fileExtension}")
+            }
+
+            if (!globalOptionsForSubcommands.forceOverwrite) {
+                val existingOutputFiles = outputFiles.filter { it.exists() }
+                if (existingOutputFiles.isNotEmpty()) {
+                    throw UsageError("None of the output files $existingOutputFiles must exist yet.", statusCode = 2)
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/EvaluatorCommand.kt:252]: This moves this check to after parsing the input file and running the evaluator, which can take some time for large result files. I prefer the command to fail early like it was before.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt [hunk 2]
```diff
@@ -205,27 +194,41 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate rules
...
+            ortResultInput = ortResultInput?.replaceConfig(it.readValue())
+        }
+
+        packageCurationsFile?.let {
+            ortResultInput = ortResultInput?.replacePackageCurations(it.readValue())
+        }
+
+        val finalOrtResult = requireNotNull(ortResultInput) {
+            "The '--ort-file' option is required unless the '--syntax-check' option is used."
+        }
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/EvaluatorCommand.kt:217]: After `requireNotNull` `ortResultInput` is smart-cast to `OrtResult`, so if you move this directly below the assignment you don't need the `?` in the other code and also don't need to assign `finalOrtResult`.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt:217]: Unfortunately, Kotlin smart casts do not work for clikt options.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt]: Done.

**Comment by sschuberth** [comment]
@mnonnenmacher if you approve, I'll leave the merge to you as I expect you need to internally account for the renaming of `--syntax-check` to `--check-syntax`.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**